### PR TITLE
Add cookie support

### PR DESCRIPTION
**Pull request by allentv**
This is an intial effort to add support for cookies. I would like to get some feedback about this approach after which I can add tests.

### CODE REVIEW

**Comment by allentv** [comment]
@googlebot I signed it!


Regards,
Allen Thomas Varghese
+353 89 965 7732

LinkedIn : http://ie.linkedin.com/in/allentv
Twitter : https://twitter.com/allentv4u

**Comment by allentv** [comment]
This PR should implement some part of https://github.com/dart-lang/shelf/issues/29

File: lib/src/request.dart [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:convert';
+import 'dart:io';
```
Comment (kevmoo) [inline lib/src/request.dart:6]: Sadly, we can't land this. We don't want `dart:io` "things" leaking into the core library. It's meant to work without `dart:io`

**Comment by kevmoo** [comment]
This approach won't work. We don't want `dart:io` types leaking into anything exposed by `shelf.dart`

File: lib/src/request.dart (hunk 1 above)
Comment (allentv) [reply lib/src/request.dart:6]: Thanks for the review. Appreciate it!

**Pull request closed without merging**